src/LetsEncrypt/Internal/CertificateSelector.cs

```diff
@@ -79,7 +79,7 @@ private static void AddWithDomainName(ConcurrentDictionary<string, X509Certifica
                 {
                     _logger.LogTrace("Using ALPN challenge cert for {domainName}", domainName);
 
-                    return challengeCert;
+                    return PreloadIntermediateCertificates(challengeCert);
```

Comment:
We can probably skip this. This code path will only return self-signed certs as a part of the TLS-ALPN-01 challenge response.